### ISSUE TITLE
installation breaks nginx

### ISSUE THREAD

**Issue by scwall** [D1ceWard/dokku-posteio#7]
**Describe the bug**
the plugin doesn't seem to work with lets encrypt ?
**To Reproduce**
I wanted to test the plugin on a new instance of a server however I have the impression that there is an incompatibility with dokku ssl
here is the log 

thanks in advance
```
root@v904:~# sudo dokku plugin:install https://github.com/dokku/dokku-letsencrypt.git
-----> Cloning plugin repo https://github.com/dokku/dokku-letsencrypt.git to /var/lib/dokku/plugins/available/letsencrypt
Cloning into 'letsencrypt'...
remote: Enumerating objects: 778, done.
remote: Counting objects: 100% (240/240), done.
remote: Compressing objects: 100% (159/159), done.
remote: Total 778 (delta 151), reused 152 (delta 80), pack-reused 538
Receiving objects: 100% (778/778), 159.66 KiB | 3.99 MiB/s, done.
Resolving deltas: 100% (478/478), done.
-----> Plugin letsencrypt enabled
v4.3.1: Pulling from goacme/lego
8464c5956bbe: Pull complete
57b6439b4724: Pull complete
7f7f62561bed: Pull complete
c824ba218e5a: Pull complete
Digest: sha256:ddb06ca69e4790c345495926ac5ff5f29bc52985e2df6a7d0026e18dc689e37e
Status: Downloaded newer image for goacme/lego:v4.3.1
docker.io/goacme/lego:v4.3.1
Adding user dokku to group adm
Starting nginx (via systemctl): nginx.service.
-----> Priming bash-completion cache
root@vm904:~# sudo dokku plugin:install https://github.com/dokku/dokku-redis.git redis
-----> Cloning plugin repo https://github.com/dokku/dokku-redis.git to /var/lib/dokku/plugins/available/redis
Cloning into 'redis'...
remote: Enumerating objects: 2332, done.
remote: Counting objects: 100% (550/550), done.
remote: Compressing objects: 100% (360/360), done.
remote: Total 2332 (delta 319), reused 366 (delta 156), pack-reused 1782
Receiving objects: 100% (2332/2332), 462.07 KiB | 7.22 MiB/s, done.
Resolving deltas: 100% (1536/1536), done.
-----> Plugin redis enabled
Adding user dokku to group adm
Starting nginx (via systemctl): nginx.service.
6.2.6: Pulling from library/redis
a2abf6c4d29d: Pull complete
c7a4e4382001: Pull complete
4044b9ba67c9: Pull complete
c8388a79482f: Pull complete
413c8bb60be2: Pull complete
1abfd3011519: Pull complete
Digest: sha256:db485f2e245b5b3329fdc7eff4eb00f913e09d8feb9ca720788059fdc2ed8339
Status: Downloaded newer image for redis:6.2.6
docker.io/library/redis:6.2.6
1.31.1-uclibc: Pulling from library/busybox
76df9210b28c: Pull complete
Digest: sha256:cd421f41ebaab52ae1ac91a8391ddbd094595264c6e689954b79b3d24ea52f88
Status: Downloaded newer image for busybox:1.31.1-uclibc
docker.io/library/busybox:1.31.1-uclibc
0.3.3: Pulling from dokku/ambassador
aad63a933944: Pull complete
2888dfab2eb5: Pull complete
51ccf60e0642: Pull complete
Digest: sha256:87c0214e190e7f6975953027157a8933701596b4b864ff66dd3cc3f6ead5c38d
Status: Downloaded newer image for dokku/ambassador:0.3.3
docker.io/dokku/ambassador:0.3.3
0.10.3: Pulling from dokku/s3backup
aad63a933944: Already exists
6654c5b7b2dc: Pull complete
26abcd9faf98: Pull complete
d1a36cd3ba61: Pull complete
9517d44e685b: Pull complete
32e8b2c4797f: Pull complete
Digest: sha256:3651f8ef12000206df55fec8ad4860d6f26b2b5af1308c0e2358253641626024
Status: Downloaded newer image for dokku/s3backup:0.10.3
docker.io/dokku/s3backup:0.10.3
0.4.3: Pulling from dokku/wait
aad63a933944: Already exists
3409ea528c35: Pull complete
88e35d065209: Pull complete
Digest: sha256:5eb9da766abdd5e8cedbde9870acd4b54c1c7e63e72c99e338b009d06f808f04
Status: Downloaded newer image for dokku/wait:0.4.3
docker.io/dokku/wait:0.4.3
-----> Priming bash-completion cache
root@vm904:~# sudo dokku plugin:install https://github.com/dokku/dokku-postgres.git postgres
-----> Cloning plugin repo https://github.com/dokku/dokku-postgres.git to /var/lib/dokku/plugins/available/postgres
Cloning into 'postgres'...
remote: Enumerating objects: 2326, done.
remote: Counting objects: 100% (387/387), done.
remote: Compressing objects: 100% (257/257), done.
remote: Total 2326 (delta 226), reused 248 (delta 108), pack-reused 1939
Receiving objects: 100% (2326/2326), 479.50 KiB | 3.50 MiB/s, done.
Resolving deltas: 100% (1562/1562), done.
-----> Plugin postgres enabled
Adding user dokku to group adm
Starting nginx (via systemctl): nginx.service.
13.4: Pulling from library/postgres
7d63c13d9b9b: Pull complete
cad0f9d5f5fe: Pull complete
ff74a7a559cb: Pull complete
c43dfd845683: Pull complete
e554331369f5: Pull complete
d25d54a3ac3a: Pull complete
bbc6df00588c: Pull complete
d4deb2e86480: Pull complete
d4132927c0d9: Pull complete
3d03efa70ed1: Pull complete
645312b7d892: Pull complete
3cc7074f2000: Pull complete
4e6d0469c332: Pull complete
Digest: sha256:1adb50e5c24f550a9e68457a2ce60e9e4103dfc43c3b36e98310168165b443a1
Status: Downloaded newer image for postgres:13.4
docker.io/library/postgres:13.4
-----> Priming bash-completion cache
root@vm04:~# sudo dokku plugin:install https://github.com/dokku/dokku-mysql.git mysql
-----> Cloning plugin repo https://github.com/dokku/dokku-mysql.git to /var/lib/dokku/plugins/available/mysql
Cloning into 'mysql'...
remote: Enumerating objects: 2143, done.
remote: Counting objects: 100% (395/395), done.
remote: Compressing objects: 100% (273/273), done.
remote: Total 2143 (delta 226), reused 243 (delta 99), pack-reused 1748
Receiving objects: 100% (2143/2143), 450.15 KiB | 4.29 MiB/s, done.
Resolving deltas: 100% (1409/1409), done.
-----> Plugin mysql enabled
8.0.27: Pulling from library/mysql
72a69066d2fe: Pull complete
93619dbc5b36: Pull complete
99da31dd6142: Pull complete
626033c43d70: Pull complete
37d5d7efb64e: Pull complete
ac563158d721: Pull complete
d2ba16033dad: Pull complete
688ba7d5c01a: Pull complete
00e060b6d11d: Pull complete
1c04857f594f: Pull complete
4d7cfa90e6ea: Pull complete
e0431212d27d: Pull complete
Digest: sha256:e9027fe4d91c0153429607251656806cc784e914937271037f7738bd5b8e7709
Status: Downloaded newer image for mysql:8.0.27
docker.io/library/mysql:8.0.27
Adding user dokku to group adm
Starting nginx (via systemctl): nginx.service.
-----> Priming bash-completion cache
root@vm04:~# sudo dokku plugin:install https://github.com/dokku/dokku-mysql.git mysql
-----> Cloning plugin repo https://github.com/dokku/dokku-mysql.git to /var/lib/dokku/plugins/available/mysql
fatal: destination path 'mysql' already exists and is not an empty directory.
2021/12/30 20:23:47 exit status 128
root@vm4:~# dokku plugin:install https://github.com/D1ceWard/dokku-posteio.git
-----> Cloning plugin repo https://github.com/D1ceWard/dokku-posteio.git to /var/lib/dokku/plugins/available/posteio
Cloning into 'posteio'...
remote: Enumerating objects: 392, done.
remote: Counting objects: 100% (121/121), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 392 (delta 83), reused 72 (delta 39), pack-reused 271
Receiving objects: 100% (392/392), 72.24 KiB | 9.03 MiB/s, done.
Resolving deltas: 100% (244/244), done.
-----> Plugin posteio enabled
Adding user dokku to group adm
Starting nginx (via systemctl): nginx.service.
2.3.4: Pulling from analogic/poste.io
8f04e8168e38: Pull complete
1cc4842b9e60: Pull complete
cce7cfd39e8b: Pull complete
b66992c2a58d: Pull complete
f3694da36af1: Pull complete
fd3e7420488d: Pull complete
f2b04b9ce10f: Pull complete
007c9376eada: Pull complete
3f1c245fe31a: Pull complete
1239ff985ebb: Pull complete
b4d72c69e2cf: Pull complete
445504d85534: Pull complete
6742cbe0b893: Pull complete
47961d41b33d: Pull complete
829a37047893: Pull complete
2a5d485e7f86: Pull complete
7785b3718560: Pull complete
946190dd4492: Pull complete
40b2581b9854: Pull complete
f083c807ad39: Pull complete
874a576a404a: Pull complete
40e302e141be: Pull complete
27935e1ca5d6: Pull complete
Digest: sha256:866de2a94198c0f82a1d1aa613528d26ca73c7f6a7107db4ce30504b74d9a329
Status: Downloaded newer image for analogic/poste.io:2.3.4
docker.io/analogic/poste.io:2.3.4
-----> Creating data directory
-----> Creating nginx conf directory
-----> Creating nginx conf files
-----> Restarting nginx
-----> Priming bash-completion cache
root@904:~# dokku posteio:set-domain mail.geekn.xyz
-----> Setup custom domain
root@404:~# dokku posteio:start
-----> Starting container
-----> Creating nginx config file
Reloading nginx configuration (via systemctl): nginx.serviceJob for nginx.service failed.
See "systemctl status nginx.service" and "journalctl -xe" for details.
 failed!
root@904:~# journalctl -xe
-- Support: http://www.ubuntu.com/support
--
-- The unit run-docker-runtime\x2drunc-moby-a2a6c78a4e5451fa995bca7209ee8327b69c7e197d657ef0af04a27fb2d4309c-runc.MJVO4L.mount has successfully entered the 'dea>
Dec 30 20:25:23 v904.****.net systemd[498]: run-docker-runtime\x2drunc-moby-a2a6c78a4e5451fa995bca7209ee8327b69c7e197d657ef0af04a27fb2d4309c-runc.>
-- Subject: Unit succeeded
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- The unit UNIT has successfully entered the 'dead' state.
Dec 30 20:25:23 v904.****.net kernel: cgroup: cgroup: disabling cgroup2 socket matching due to net_prio or net_cls activation
Dec 30 20:25:23 v94.****.net kernel: eth0: renamed from veth505c279
Dec 30 20:25:23 v44.****.net systemd-networkd[306]: veth7643134: Gained carrier
Dec 30 20:25:23 v44.****.net systemd-networkd[306]: docker0: Gained carrier
Dec 30 20:25:23 vm94.****.net kernel: docker0: port 1(veth7643134) entered blocking state
Dec 30 20:25:23 vm04.****.net kernel: docker0: port 1(veth7643134) entered forwarding state
Dec 30 20:25:23 vmi04.****.net sudo[25211]:    dokku : TTY=pts/0 ; PWD=/root ; USER=root ; COMMAND=/etc/init.d/nginx reload
Dec 30 20:25:23 vmi94.****.net sudo[25211]: pam_unix(sudo:session): session opened for user root by root(uid=0)
Dec 30 20:25:23 vmi04.****.net systemd[1]: Reloading A high performance web server and a reverse proxy server.
-- Subject: A reload job for unit nginx.service has begun execution
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- A reload job for unit nginx.service has begun execution.
--
-- The job identifier is 1159.
Dec 30 20:25:23 vmi04.****.net nginx[25315]: nginx: [emerg] no "ssl_certificate" is defined for the "listen ... ssl" directive in /var/lib/dokku/da>
Dec 30 20:25:23 vmi04.****.net systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
-- Subject: Unit process exited
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- An ExecReload= process belonging to unit nginx.service has exited.
--
-- The process' exit code is 'exited' and its exit status is 1.
Dec 30 20:25:23 vmi04.****.net systemd[1]: Reload failed for A high performance web server and a reverse proxy server.
-- Subject: A reload job for unit nginx.service has finished
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- A reload job for unit nginx.service has finished.
--
-- The job identifier is 1159 and the job result is failed.
Dec 30 20:25:23 vmi04.****.net sudo[25211]: pam_unix(sudo:session): session closed for user root
Dec 30 20:25:23 vmi04.****.net sudo[24879]: pam_unix(sudo:session): session closed for user dokku
```
```

**Comment by D1ceWard** [comment]
Firstly, thank you for the complete list of the different steps that led to the error.
The bug comes from the way the plugin handles SSL, nothing to do with letsencrypt as it is totally independent.

Nginx doesn't seem to like having the ssl directive without having a certificate, I don't know why I never had the problem at first installation knowing that I've had several instances working for a while so maybe nginx, in the meantime, has made an update to avoid the ssl directive without a certificate.

From nginx.conf.sigil file
```
server {
  # ssl directive on listens are always here
  listen [::]:443 ssl http2;
  listen 443 ssl http2; 

  ...

  # custom_ssl.conf file is only there when you activate ssl with
  # `dokku posteio:toggle-ssl` and contain certificate path
  include {{ $.POSTEIO_ROOT }}/custom_ssl.conf*;
```

Example content of custom_ssl.conf 
```
ssl_certificate ${POSTEIO_ROOT}/data/ssl/server.crt;
ssl_certificate_key ${POSTEIO_ROOT}/data/ssl/server.key;
ssl_protocols TLSv1.2;
ssl_prefer_server_ciphers on;" > "${POSTEIO_ROOT}/custom_ssl.conf
```

custom_ssl.conf file exists because SSL is managed internally by Poste.io

![image](https://user-images.githubusercontent.com/8332823/148200872-f700c2d3-bf19-4b17-899e-8015c718f9a0.png)

Now that I fully understand the problem I'll do a fix as soon as possible.

**Comment by scwall** [comment]
Perfect, thank you for your answer

**Comment by D1ceWard** [comment]
I have just released version v1.6 which fixes the problem :sunglasses:
https://github.com/D1ceWard/dokku-posteio/releases

Do not hesitate to open another issue if other problems arise.
